### PR TITLE
test: add --block-domains integration tests

### LINKED ISSUE #1041: test: add --block-domains integration tests
## Parent Issue
Part of #1039

## Problem
The \`--block-domains\` CLI flag (deny-list on top of allow-list) has zero integration tests. The \`blocked-domains.test.ts\` file is a misnomer — it only tests allow-list behavior. The \`AwfRunner\` test fixture doesn't even expose a \`blockDomains\` option.

## Acceptance Criteria
- [ ] Add \`blockDomains\` option to \`AwfRunner\` in \`tests/fixtures/awf-runner.ts\` (maps to \`--block-domains\` CLI flag)
- [ ] Add integration tests in \`tests/integration/blocked-domains.test.ts\` (or a new file) covering:
  - Block specific subdomain while allowing parent domain (\`--allow-domains github.com --block-domains api.github.com\`)
  - Block takes precedence over allow
  - Wildcard blocking patterns (\`--block-domains *.evil.com\`)
  - Multiple blocked domains
- [ ] Tests should use \`toBlockDomain()\` / \`toFail()\` matchers where appropriate

## References
- \`src/cli.ts\` line 634: \`--block-domains\` flag definition
- \`src/squid-config.ts\`: How blocked domains are added to Squid ACL
- \`tests/fixtures/awf-runner.ts\`: \`AwfOptions\` interface (needs \`blockDomains\` added)

Closes #1039 (when combined with other sub-issues)

### PR DESCRIPTION
**Pull request by Mossaka**
## Summary
- Add `blockDomains?: string[]` option to `AwfRunner` test fixture in `tests/fixtures/awf-runner.ts`
- Add `--block-domains` flag handling in both `run()` and `runWithSudo()` methods
- Add 6 integration tests for the `--block-domains` deny-list feature in `tests/integration/blocked-domains.test.ts`

## Test Coverage
- Block specific subdomain while allowing parent domain (`--allow-domains github.com --block-domains api.github.com`)
- Allow non-blocked subdomains when parent is allowed
- Block takes precedence over allow (same domain in both lists)
- Wildcard blocking patterns (`--block-domains *.github.com`)
- Multiple blocked domains
- Debug output shows blocked domains configuration

## Test plan
- [x] TypeScript compiles cleanly
- [x] `npm run lint` passes (0 errors)
- [x] `npm run build` passes
- [x] Integration tests verified locally (3/6 passed, 3 failed due to Docker container conflicts from concurrent test runs - not related to test logic)

Closes #1041

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/integration/blocked-domains.test.ts [hunk 1]
```diff
@@ -183,3 +183,96 @@ describe('Domain Allowlist Edge Cases', () => {
...
+      {
+        allowDomains: ['github.com'],
+        blockDomains: ['api.github.com', 'raw.githubusercontent.com'],
+        logLevel: 'debug',
+        timeout: 60000,
+      }
+    );
+    expect(result.stdout).toContain('exit=');
+    expect(result.stdout).not.toContain('exit=0');
+  }, 120000);
```
[the remark this answers is not in the document]
Comment (Mossaka) [reply tests/integration/blocked-domains.test.ts:267]: Fixed in 85e27f7. Updated the test to curl both `api.github.com` and `raw.githubusercontent.com`, and added `githubusercontent.com` to allowDomains so the blocklist (not the allowlist) is what causes the failures.
